e2e/brickmux_test.go

```diff
@@ -276,5 +276,176 @@ func TestBrickMux(t *testing.T) {
 		r.Nil(client.VolumeStop(volname1 + strconv.Itoa(i)))
 		r.Nil(client.VolumeDelete(volname1 + strconv.Itoa(i)))
 	}
+
+	// Turn on brick mux cluster option
+	optReq = api.ClusterOptionReq{
+		Options: map[string]string{"cluster.max-bricks-per-process": "2"},
+	}
+	err = client.ClusterOptionSet(optReq)
+	r.Nil(err)
+
+	for i := 37; i <= 54; i++ {
+		brickPath := testTempDir(t, "brick")
+		brickPaths = append(brickPaths, brickPath)
+	}
+
+	/* Test  for Testing max-bricks-per-process constraint while
+	multiplexing*/
+
+	// Create 5 volumes and start all 5
+	// making all brick multiplexed with the constraint that
+	// max-bricks-per-process is 2
+
+	index = 37
+	for i := 1; i <= 5; i++ {
+		createReq := api.VolCreateReq{
+			Name: volname1 + strconv.Itoa(i),
+			Subvols: []api.SubvolReq{
+				{
+					Type: "distribute",
```

Comment:
I would like to see tests for different volume types.

Reply:
Done